tests/ui/manual_let_else.rs

```diff
@@ -248,4 +254,13 @@ fn not_fire() {
         Some(value) => value,
         _ => macro_call!(),
     };
+
+    fn e() -> Variant {
```

Comment:
Just a note: this is inside a function named `not_fire`. At least if you follow how I have originally designed the file,
the test should be moved to the end of the function `do_fire` (as should the Issue 9440 snippet above).

Maybe clippy conventions are different and this design is unique among clippy, idk. I did it back then because it seemed to me that this design makes it very easy to verify that the `not_fire` section does indeed not fire, you just look at the last emitted lint's line number and see that it is prior to the line number of the `not_fire` function. I'd have mixed firing and not firing had clippy included `ERROR` lines as the rustc testsuite does (which is IMO a good idea in general because in clippy you often have `// should fire` or `// should not fire` comments that do the same but are not machine verified).

Reply:
I like your design. (Sorry for bothering, I've just overlooked it, I usually add new specs at the end of files.)
I edited the last commit and moved them into `fire()`.
@est31 Thank you for your review:)